deploy/helm-chart/templates/deployment.yaml

```diff
@@ -0,0 +1,97 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "nginx-gateway.fullname" . }}
+  labels:
+  {{- include "nginx-gateway.labels" . | nindent 4 }}
+spec:
+  # We only support a single replica for now
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx-gateway
+    {{- include "nginx-gateway.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        app: nginx-gateway
+      {{- include "nginx-gateway.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+      - args:
+        - static-mode
+        - --gateway-ctlr-name=k8s-gateway.nginx.org/nginx-gateway-controller
```

Comment:
do we support installing multiple releases (copies) of NKG via Helm? If we do, we need to support configuring `--gateway-ctlr-name=`. Because NKG will process and update all GatewayClasses that belong to  `k8s-gateway.nginx.org/nginx-gateway-controller` controllerName.